lib/MetadataStore.js

```diff
@@ -371,6 +371,61 @@ MetadataStore.prototype = {
   }),
 
   /**
+   * Get page metadata (including images) for the given cache_keys.
+   * For the missing cache_keys, it simply ignores them and will not
+   * raise any exception
+   *
+   * @param {Array} cacheKeys an cache key array
+   *
+   * Returns a promise with the array of retrieved metadata records
+   */
+  asyncGetMetadataByCacheKey: Task.async(function*(cacheKeys) {
+    const quoted = cacheKeys.map(key => {return `'${key}'`;}).join(",");
+
+    let metaObjects = yield this.asyncExecuteQuery(
+      `SELECT * FROM page_metadata WHERE cache_key IN (${quoted})`,
```

Comment:
I haven't done any query benchmark yet. But I did try the one gigantic query, the problem was that you had to sort out the flattened many-to-many result afterward. For instance,

```
metadata1, image1
metadata1, image2
metadata1, favicon1
metadata1, favicon2
...
metadataN, image1
...
metadataN, imageN
```

I think I just chose the simplicity other than performance for now.